[WIP] Add Drupal 8

fureigh opened the pull request:
DO NOT MERGE.

### Known TODOs:
* [ ] Update README to account for both Drupal 7 and 8.
* [ ] Finish solving deployment issue. (Currently it's possible to get partway through an install but then it fails.)

fureigh committed:
Initial commit for Drupal 8

fureigh committed:
Move PHP extensions from JSON file to php.ini file

PHP_EXTENSIONS has been deprecated, so this commit reworks the
way that we specify the extensions to include.

fureigh committed:
Change database collation to utf8mb4_general_ci

This change makes it possible to work with an AWS MySQL service.

fureigh committed:
Turn on buildpack debugging, refactor file organization

fureigh committed:
Add PHP extensions

jmcarp commented on drupal-7/pipeline.yml:
```diff
@@ -0,0 +1,50 @@
+jobs:
+- name: deploy-drupal-7
+  plan:
+  - aggregate: *plan-gets
```
This reference doesn't have a corresponding anchor, so it's not valid.

fureigh replied:
Thanks! I hadn't meant to include that particular file; I've removed it now and will finish it up in a different branch.

jmcarp commented on drupal-7/pipeline.yml:
```diff
@@ -0,0 +1,50 @@
+  on_failure:
+    put: slack
+    params:
+      <<: *slack-params
```
Same as above--there's no `&slack-params` anchor to reference.

fureigh committed:
Move Drupal 7 files to their own folder

As preparation for adding Drupal 8 files, this commit simply moves
Drupal 7 files into their own folder. Deployment instructions will
need to be updated accordingly.

fureigh committed:
Move Drupal 8 files to their own folder

A la 843510854e3c83eff54ad0caf14b80695666683a.

pburkholder commented:
Closing this since superseded by #12
